x-pack/plugin/eql/src/main/java/org/elasticsearch/xpack/eql/execution/sequence/TumblingWindow.java

```diff
@@ -511,7 +707,7 @@ private void setupWindowFromTail(Ordinal from) {
                 until.queryRequest().from(from).nextAfter(from);
             }
             // reset all sub queries
-            for (int i = 2; i < maxStages; i++) {
+            for (int i = nextPositiveStage(matcher.firstPositiveStage()); i < maxStages; i++) {
```

Comment:
Is this accurate? The original algorithm used the first criterion as the base query sorted descending, while the rest of the criteria sorted ascending. The original algorithm reset criteria starting with 2, in your case you start from the same positive stage not from the next next positive stage.